enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,350 @@
+| cluster-ingress-operator | DNS records |
+| machine-api-provider-azure | Application Security Group, Availability Set, Group, Load Balancer, Public IP Address, Route, Network Security Group, Virtual Machine Extension, Virtual Interface, Virtual Machine, Virtual Network. |
+
+Below list of terraform Azure APIs to create resources should be updated to add user
+defined tags and as well the openshift default tag in the installer component.
+`azurerm_resource_group, azurerm_image, azurerm_lb, azurerm_network_security_group, azurerm_storage_account, azurerm_user_assigned_identity, azurerm_virtual_network, azurerm_linux_virtual_machine, azurerm_network_interface, azurerm_dns_cname_record`
+
+API update example:
+A local variable should be defined, which merges the default tag and the user
+defined Azure tags, which should be referenced in the Azure resource APIs.
+``` terraform
+locals {
+  tags = merge(
+    {
+      "kubernetes.io_cluster.${var.cluster_id}" = "owned"
+    },
+    var.azure_extra_tags,
+  )
+}
+
+resource "azurerm_resource_group" "main" {
+  tags     = local.tags
+}
+```
+
+The tag of the form `kubernetes.io_cluster.<cluster_id>:owned` added by openshift, where
+cluster_id is a string generated by concatenating user inputted cluster name and a random 
+string will be limited to a maximum length of 27 characters by trimming long cluster name 
+to 21 characters.
+
+#### Caveats
+1. Updating or removing resource tags added by Openshift, using an external interface
+   may or may not be reconciled by the operator managing the resource.
+2. Updating tags of individual resources is not supported and any tag present in 
+   `.status.platformStatus.azure.resourceTags` of `infrastructure.config.openshift.io/v1` resource 
+   will result in adding tags to all Openshift managed Azure resources. 
+
+### Risks and Mitigations
+
+### Drawbacks
+- User-defined tags cannot be updated on an Azure resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in the scope.
+  User-defined tags can be updated on the following Azure resources.
+  - ResourceGroup
+  - Storage Account
+  - DNS Zones
+  - DNS Records
+  - ApplicationSecurityGroup
+  - AvailabilitySet
+  - Group
+  - LoadBalancer
+  - PublicIPAddress
+  - RouteTable
+  - SecurityGroup
+  - VirtualMachineExtension
+  - NetworkInterface
+  - VirtualMachine
+  - VirtualNetwork
+
+- Because tags in `userTags` must be applied to all Azure resources that OpenShift creates,
+  tags must fit all the constraints of the various Azure resources as described below:
+    1. Tag names cannot have `microsoft`, `azure`, `windows` prefixes which are
+       reserved for Azure use.
+    2. An Azure storage account has a limit of 128 characters for the tag name.
+    3. An Azure DNS zone or Traffic or Front Door resource tag name cannot have spaces,
```

Comment:
According to [Microsoft's docs](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/tag-support), "Front Door doesn't support the use of # or : in the tag name."

Reply:
Yes, tag names cannot contain those characters and is restricted in OCP as well. It's captured [here](https://github.com/openshift/enhancements/pull/1191/files/47f5b58a740b0f8a16c98baf5af899b7555c59a4#diff-07e0630cc477ba74bc69ea6a15ecefd4f5ce32927210e4e7b0b89d423d2193dbR78).